src/material/datepicker/month-view.ts

```diff
@@ -252,6 +252,19 @@ export class MatMonthView<D> implements AfterContentInit, OnChanges, OnDestroy {
     this._changeDetectorRef.markForCheck();
   }
 
+  _dateBecomesActive(event: MatCalendarUserEvent<number>) {
+    const date = event.value;
+    const activeYear = this._dateAdapter.getYear(this.activeDate);
+    const activeMonth = this._dateAdapter.getMonth(this.activeDate);
+    const activeDate = this._dateAdapter.createDate(activeYear, activeMonth, date);
```

Comment:
Is this all still necessary? `getYear`, `getMonth`, etc. all take a date `D`, which you're then composing into... another date `D`. I think that the other function you were referencing did something like this to deal with ranges. You should be able to just do
```ts
    if (!this._dateAdapter.sameDate(event.value, this._activeDate)) {
      this.activeDateChange.emit(activeDate);
    }
```

We should also confirm that the `markForCheck` is actually necessary here

Reply:
yes, constructing a date `D` from the event is really needed. That's because the event.value is a number that corresponds to the day of the month that becomes active. We turn that into a date `D` relative to the month and year of the current date.

The code from `_dateSelected` that references `rangeStartDate` and `rangeEndDate` is not needed here because activeDate doesn't needed to deal with ranges, but we really do need to convert the `MatCalendarUserEvent<number>` into a date `D`.

Reply:
I don't think the `markForCheck` is necessary here because this method doesn't change any instance values on the component, so change detection doesn't seem relevant.

Reply:
Since the month and year are the same, though, can't you just compare the dates, then?

Reply:
Yeah, that should work since we're not actually assigning to `activeDate`. I was confused between this method and `_handleCalendarBodyKeydown`.

Reply:
We can do that for the check if the active date changed, but we still need a date `D` to pass to `activeDateChange`.

Reply:
Ack, that makes sense